src/cloudsc_fortran_atlas/cloudsc_global_atlas_state_mod.F90

```diff
@@ -190,73 +192,136 @@ SUBROUTINE CLOUDSC_GLOBAL_ATLAS_STATE_BLOCK(SELF, FSET, IBLK)
     CALL FSET%DATA(40, SELF%PFCQSNG, IBLK)
     CALL FSET%DATA(41, SELF%PFSQLTUR, IBLK)
     CALL FSET%DATA(42, SELF%PFSQITUR, IBLK)
-    CALL FSET%DATA(43,SELF%PFPLSL, IBLK)
+    CALL FSET%DATA(43, SELF%PFPLSL, IBLK)
     CALL FSET%DATA(44, SELF%PFPLSN, IBLK)
     CALL FSET%DATA(45, SELF%PFHPSL, IBLK)
     CALL FSET%DATA(46, SELF%PFHPSN, IBLK)
     CALL FSET%DATA(47, SELF%PCOVPTOT, IBLK)
     CALL FSET%DATA(48, SELF%PRAINFRAC_TOPRFZ, IBLK)
   END SUBROUTINE CLOUDSC_GLOBAL_ATLAS_STATE_BLOCK
 
-  SUBROUTINE CLOUDSC_GLOBAL_ATLAS_STATE_LOAD(SELF, FSET, NPROMA, NGPTOTG)
+  SUBROUTINE CLOUDSC_GLOBAL_ATLAS_STATE_LOAD(SELF, FSET, FSPACE, NPROMA, NGPTOTG)
     ! Load reference input data via serialbox
     CLASS(CLOUDSC_GLOBAL_ATLAS_STATE) :: SELF
     TYPE(ATLAS_FIELDSET), INTENT(INOUT) :: FSET
-    INTEGER(KIND=JPIM), INTENT(IN) :: NPROMA
-    INTEGER(KIND=JPIM), INTENT(IN), OPTIONAL :: NGPTOTG
+    TYPE(ATLAS_FUNCTIONSPACE_BLOCKSTRUCTUREDCOLUMNS), INTENT(INOUT) :: FSPACE
+    INTEGER(KIND=JPIM), INTENT(IN) :: NPROMA, NGPTOTG
 
     TYPE(ATLAS_STRUCTUREDGRID) :: GRID
-    TYPE(ATLAS_FUNCTIONSPACE_BLOCKSTRUCTUREDCOLUMNS) :: FSPACE
+    TYPE(ATLAS_MULTIFIELD) :: MULTIFIELD_IN, MULTIFIELD_OUT
     INTEGER(KIND=JPIM) :: KLON, IVAR, B
-    TYPE(VAR3D_PTR), DIMENSION(24) :: IN_VARS_3D_REAL64
-    TYPE(VAR3D_PTR), DIMENSION(15) :: OUT_VARS_3D_REAL64
     REAL(KIND=JPRB), POINTER :: TMP3D(:,:,:)
     REAL(KIND=JPRB), POINTER :: TMP2D(:,:)
     TYPE(ATLAS_FIELD) :: FIELD
-    TYPE(ATLAS_PARTITIONER) :: PARTITIONER
-    INTEGER(KIND=JPIM)      :: NGPTOT
+    TYPE(ATLAS_CONFIG) :: CONFIG
+    TYPE(ATLAS_CONFIG), DIMENSION(24) :: IN_MFIELD_CONFIG ! the last five variables are special and added through FieldSet
+    TYPE(ATLAS_CONFIG), DIMENSION(14) :: OUT_MFIELD_CONFIG ! the last two variables are special and added through FieldSet
+    INTEGER :: INVAR_SIZE, OUTVAR_SIZE
+    LOGICAL :: LMULTIFIELD
+    CHARACTER(len=8) :: CENV, FPREC
+    INTEGER :: CENV_LEN
+    TYPE(ATLAS_TRACE) :: TRACE, TRACE_IO
+
+    TRACE = ATLAS_TRACE("cloudsc_global_atlas_state_mod.F90", __LINE__, &
+        & "CLOUDSC_GLOBAL_ATLAS_STATE_LOAD")
+
+    LMULTIFIELD = .TRUE.
+    CALL GET_ENVIRONMENT_VARIABLE("CLOUDSC_ATLAS_MULTIFIELD",CENV,CENV_LEN)
+    IF (CENV_LEN > 0 ) THEN
+      IF( TRIM(CENV) == "0" .OR. TRIM(CENV) == "OFF" .OR. TRIM(CENV) == "FALSE" ) THEN
+        LMULTIFIELD = .FALSE.
+      ENDIF
+    ENDIF
```

Comment:
I just noticed that we don't have a description of this variant in the README. Could you add just 2-3 sentences that explain what this variant is (e.g., "uses Atlas as storage backend for field data") and that two choices for the backend exist Multifield/FieldSet, which can be switched using that environment variable.

Reply:
Good point! Added 2-3 sentences to README. Thanks for the review!